core/java/iesi-rest-without-microservices/src/main/java/io/metadew/iesi/server/rest/executionrequest/ExecutionRequestDtoRepository.java

```diff
@@ -22,317 +23,159 @@
+                    } else if (executionRequestFilter.getExecutionRequestFilterOption().equals(ExecutionRequestFilterOption.ENVIRONMENT)) {
+                        return " script_execution_requests.ENVIRONMENT " + (executionRequestFilter.isExactMatch() ? "=" : "LIKE") + " '" + (executionRequestFilter.isExactMatch() ? "" : "%") + executionRequestFilter.getValue() + (executionRequestFilter.isExactMatch() ? "" : "%") + "' ";
+                    } else if (executionRequestFilter.getExecutionRequestFilterOption().equals(ExecutionRequestFilterOption.ID)) {
+                        return " execution_requests.REQUEST_ID " + (executionRequestFilter.isExactMatch() ? "=" : "LIKE") + " '" + (executionRequestFilter.isExactMatch() ? "" : "%") + executionRequestFilter.getValue() + (executionRequestFilter.isExactMatch() ? "" : "%") + "' ";
+                    } else if (executionRequestFilter.getExecutionRequestFilterOption().equals(ExecutionRequestFilterOption.LABEL)) {
+                        return " execution_request_labels.NAME = '" + executionRequestFilter.getValue().split(":")[0] +
+                                "' and execution_request_labels.VALUE " + (executionRequestFilter.isExactMatch() ? "=" : "LIKE") + " '" + (executionRequestFilter.isExactMatch() ? "" : "%") + executionRequestFilter.getValue().split(":")[1] + (executionRequestFilter.isExactMatch() ? "" : "%") + "' ";
+                    } else {
+                        return null;
+                    }
+                }
+        )
+                .filter(Objects::nonNull)
+                .collect(Collectors.joining(" and "));
+        if (filterStatements.isEmpty()) {
+            return "";
+        }
+        return " WHERE " + filterStatements;
+    }
 
-    @Autowired
-    ExecutionRequestDtoRepository(MetadataRepositoryConfiguration metadataRepositoryConfiguration) {
-        this.metadataRepositoryConfiguration = metadataRepositoryConfiguration;
+    private String getOrderByClause(Pageable pageable) {
+        if (pageable.getSort().isUnsorted()) return " ";
+        List<String> sorting = pageable.getSort().stream().map(order -> {
+            // add further sort on the ScriptAndScriptVersionTable here
+            if (order.getProperty().equalsIgnoreCase("SCRIPT")) {
+                return "name_script_execution_requests.SCRPT_NAME " + order.getDirection();
+            } else if (order.getProperty().equalsIgnoreCase("REQUEST_TIMESTAMP")) {
+                return "execution_requests.REQUEST_TMS " + order.getDirection();
+            } else if (order.getProperty().equalsIgnoreCase("VERSION")) {
+                return "name_script_execution_requests.SCRPT_VRS " + order.getDirection();
+            } else {
+                return null;
+            }
+        })
+                .filter(Objects::nonNull)
+                .collect(Collectors.toList());
+        if (sorting.isEmpty()) {
```

Comment:
It would be better to always have a default sorting for the Front-End, e.g by ID and ASC

Reply:
when no sorting is provided, pagination is indeed not deterministic. We can have a discussion on whether or not we should make it order by default on ID or let it up to the user to decide. I created a dedicated issue for this #201